tdmpc2/tdmpc2.py

```diff
@@ -299,30 +303,48 @@ def update(self, buffer):
             if self.cfg.multitask:
                 task = task.flatten(1, 2)
 
+        # # Compute targets
+        # with torch.no_grad():
+        #     next_z = self.model.encode(obs[1:], task)
+        #     td_targets = self._td_target(next_z, reward, task)
+
+        # # Prepare for update
+        # self.optim.zero_grad(set_to_none=True)
+        # self.model.train()
+
+        # # Latent rollout
+        # zs = torch.empty(
+        #     self.cfg.horizon + 1,
+        #     self.cfg.batch_size,
+        #     self.cfg.latent_dim,
+        #     device=self.device,
+        # )
+        # z = self.model.encode(obs[0], task)
+        # zs[0] = z
+        # consistency_loss = 0
+        # for t in range(self.cfg.horizon):
+        #     z = self.model.next(z, action[t], task)
+        #     consistency_loss += F.mse_loss(z, next_z[t]) * self.cfg.rho**t
+        #     zs[t + 1] = z
         # Compute targets
         with torch.no_grad():
+            zs = self.model.encode(obs, task)
```

Comment:
I suspect this line here is your culprit. Because its wrapped in a `torch.no_grad()` it means that your line is equivalent to `zs = self.model.encode(obs,task).detach()`. In other words, your encoder is not getting trained since no gradients flow through it anymore. 

Try putting this outside the `with`